src/beamlime/config/builders.py

```diff
@@ -148,109 +87,116 @@ def build_offline_fake2d_config():
+        _app_subscription("visualisation", "reduced-data"),
     ]
+
     return config
 
 
-def _fake_event_data_stream_config() -> dict:
-    from beamlime.handlers.data_feeder import FakeEventDataFeeder as data_feeder  # TODO
-    from beamlime.handlers.data_generator import (
-        FakeDreamDataGenerator as data_generator,
-    )
+def _kafka_recipe(
+    name: str, topic: str, role: Literal["consumer", "producer"]
+) -> ServiceRecipe:
+    if role == "consumer":
+        from ..communication.interfaces import KafkaConsumer as constructor
+        from ..resources.templates import (
+            load_kafka_consumer_specs_tpl as load_specs_tpl,
+        )
+    elif role == "producer":
+        from ..communication.interfaces import KafkaProducer as constructor
+        from ..resources.templates import (
+            load_kafka_producer_specs_tpl as load_specs_tpl,
+        )
+    else:
+        raise ValueError("Not supported kafka role: ", role)
 
-    config = dict()
-    app_names = ["data-generator", "data-feeder"]
-    # Placeholders
-    config["applications"] = [
-        _default_application_config(app_name) for app_name in app_names
-    ]
-    config["application-specs"] = {app_name: {} for app_name in app_names}
+    specs = load_specs_tpl()
+    specs["topic"] = topic
 
-    # Application configurations
-    # # Data Generator
-    config["applications"][0]["data-handler"] = ".".join(
-        (data_generator.__module__, data_generator.__name__)
-    )
-    config["applications"][0]["timeout"] = 5
-    config["applications"][0]["wait-interval"] = 0.1
-    # # Data Feeder
-    config["applications"][1]["data-handler"] = ".".join(
-        (data_feeder.__module__, data_feeder.__name__)
-    )
-    config["applications"][1]["timeout"] = 10
-    config["applications"][1]["wait-interval"] = 0.5
-
-    # Application specs
-    config["application-specs"]["data-generator"] = {
-        "pulse_rate": 100_000,
-        "original_file_name:": "DREAM_baseline_all_dets.nxs",
-        "keep_tmp_file": False,
-        "num_frame": 1_000_000,
-        "max_events": 10e8,
-        "random_seed": 0,
-    }
-    config["application-specs"]["data-feeder"] = {"chunk_size": 16}
-
-    # Communication
-    channel_specs = [
-        ("kafka-producer", "KAFKA-PRODUCER", {}),
-        ("kafka-log-consumer", "KAFKA-CONSUMER", {"topics": ["log"]}),
-        ("kafka-data-consumer", "KAFKA-CONSUMER", {"topics": ["data"]}),
-    ]
-    config["communication-channels"] = [
-        _channel_config(ch_name, ch_type, **others)
-        for ch_name, ch_type, others in channel_specs
-    ]
-    config["application-subscriptions"] = [
-        _app_subscription("data-generator", "kafka-producer"),
-        _app_subscription("data-feeder", "kafka-log-consumer", "kafka-data-consumer"),
-    ]
-    return config
+    return ServiceRecipe(name, constructor, specs)
 
 
-def build_fake_event_kafka_config():
+def _fake_event_data_stream_specs() -> dict:
```

Comment:
It seems odd to have this fake-creation logic (similar above), including the configuration, mixed in with all the other code. Would it make sense to separate this more?